lib/periodicChecks.js

```diff
@@ -0,0 +1,145 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Handler for checks to be carried out periodically.
+ */
+
+const {
+  getAllOpenPullRequests,
+  hasPullRequestBeenApproved,
+  getLastReviewOfType,
+  doesPullRequestHaveChangesRequested,
+  pingAndAssignReviewers
+} = require('./utils');
+const mergeConflictModule = require('./checkMergeConflicts');
+const pullRequestReviewModule = require('./checkPullRequestReview');
+const { teamLeads } = require('../userWhitelist.json');
+const CHANGES_REQUESTED = 'changes_requested';
+const { oppiaMaintainers } = require('../userWhitelist.json');
+
+/**
+ *  This function checks if a pull request has merge conflicts.
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const hasMergeConflict = (pullRequest) => {
+  return pullRequest.mergeable === false;
+};
+
+/**
+ *  This function checks if a pull request has pending reviews.
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const hasPendingReviews = (pullRequest) => {
+  return pullRequest.requested_reviewers.length > 0;
+};
+
+/**
+ * This function ensures that a pull request is assigned.
+ * 1. Fetch all pull requests.
+ * 2. Filter out those without assignees.
+ * 3. Find out appropriate assignee.
+ *  - If pull request has a merge conflict    -- PR author.
+ *  - If pull request has pending reviews     -- Remaining reviewers.
+ *  - If pull request has changes requested   -- PR author.
+ *  - If pull request has been approved       -- Any member of the Organisation.
+ *  - If none of the above cases              -- Welfare team lead.
+ * 4. Assign appropriate assignee.
+ * 5. Ping appropriate assignee.
+ * @param {import('probot').Context} context
+ */
+const ensurePullRequestIsAssigned = async (context) => {
+  const allOpenPullRequests = await getAllOpenPullRequests(context);
+  const unassignedPullRequests = allOpenPullRequests.filter((pullRequest) => {
+    return pullRequest.assignees.length === 0;
+  });
+
+  unassignedPullRequests.forEach(async (pullRequest) => {
+    const pullRequestResponse = await context.github.pulls.get(
+      context.repo({
+        pull_number: pullRequest.number,
+      })
+    );
+    const pullRequestData = pullRequestResponse.data;
+
+    const hasBeenApproved = await hasPullRequestBeenApproved(
+      context,
+      pullRequestData.number
+    );
+    const hasChangesRequested = await doesPullRequestHaveChangesRequested(
+      context,
+      pullRequestData.number
+    );
+
+    if (hasMergeConflict(pullRequestData)) {
+      console.log('PULL REQUEST HAS MERGE CONFLICT...');
+      await mergeConflictModule.checkMergeConflictsInPullRequest(
+        context,
+        pullRequestData
+      );
+    } else if (hasPendingReviews(pullRequestData)) {
```

Comment:
In this case, we should also add a comment with list of reviewers who haven't responded within 24 hours and ping one of the core-maintainers based on a rotation whitelist (per month: one core maintainer). See the thread here: https://docs.google.com/document/d/1m-7dZWGw77hZ6GOnjBu4p1XGXxyBPY2n2pzf-76RuTY/edit?disco=AAAAJ4jo7FM

Reply:
This is not a part of GSoC - will be taken up separately after GSoC. We have a tracking issue: #118

Reply:
Thanks, @ankita240796.